test/docker-e2e/e2e_test.go

```diff
@@ -41,6 +44,61 @@ func (s *CelestiaTestSuite) SetupSuite() {
 	s.client, s.network = tastoradockertypes.DockerSetup(s.T())
 }
 
+// ExecuteNodeCommand executes a command on a chain node with common parameters automatically added.
+// This reduces boilerplate in tests by setting common flags like --home, --node, --fees, etc.
+func (s *CelestiaTestSuite) ExecuteNodeCommand(ctx context.Context, chainNode tastoratypes.ChainNode, cmd ...string) (string, string, error) {
+	s.Require().Greater(len(cmd), 0, "command must not be empty")
+
+	var finalCmd []string
+
+	if cmd[0] == "celestia-appd" {
+		s.Require().Greater(len(cmd), 1, "celestia-appd must have at least one subcommand")
+		cmd = cmd[1:]
+	}
```

Comment:
[not blocking]

I'm confused by this logic. why are we removing celestia-appd and then prepending it? why are we checking for length only if the first command is celestia-appd?

Reply:
I believe this to to basically allow for usage of the exec command with or without the binary included, could make sense to just force it one way or the other and assert its absence/presence.

Reply:
in the following lines:

```go
isTxCommand := cmd[0] == "tx"
isKeysCommand := cmd[0] == "keys"
```

We check these conditions to add some arguments in case they are not passed by the user.
If `celestia-appd` is passed by the user we have to add more conditions to check the second argument and etc...

To avoid multiple checks, we remove it if it exist, do the checks and add it at the end. so the code is cleaner this way.

We have to check if the length is greater than one if the first arg is `celestia-appd` to avoid referencing to a non existing index in the cmd slice.